kolibri/plugins/coach/assets/src/modules/pluginModule.js

```diff
@@ -106,12 +106,11 @@ export default {
         ]).catch(error => {
           store.dispatch('handleError', error);
         });
-      } else {
-        // otherwise refresh but don't block
-        return store
-          .dispatch('classSummary/loadClassSummary', classId)
-          .catch(error => store.dispatch('handleApiError', error));
       }
+      // navigating within the same class, periodically refresh but don't block
+      return store
```

Comment:
The other issue here is that returning the promise does seem to block navigation until the Promise is resolved.